aten/src/ATen/cuda/CUDAGraph.cpp

```diff
@@ -290,7 +289,7 @@ void CUDAGraph::reset() {
   // a Juptyer notebook, I don't see an easy way for reset() to gracefully fix all such possible error states.
   if (has_graph_ || has_graph_exec_) {
     // notifyCaptureDestroy may throw. How should we handle this?
-    c10::cuda::CUDACachingAllocator::notifyCaptureDestroy(capture_dev_, mempool_id_);
+    c10::cuda::CUDACachingAllocator::destroyPool(capture_dev_, mempool_id_);
```

Comment:
IMO, `destroyPool` is a little misleading. We're notifying that the current capture is destroyed, but there might be other references to the pool that keep it alive.